backend/module/eCampCore/src/Acl/UserIsAuthenticatedUser.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace eCamp\Core\Acl;
+
+use eCamp\Core\Entity\User;
+use Laminas\Authentication\AuthenticationService;
+use Laminas\Permissions\Acl\Acl;
+use Laminas\Permissions\Acl\Assertion\AssertionInterface;
+use Laminas\Permissions\Acl\Resource\ResourceInterface;
+use Laminas\Permissions\Acl\Role\RoleInterface;
+
+class UserIsAuthenticatedUser implements AssertionInterface {
+    private $authUserId;
+
+    public function assert(
+        Acl $acl,
+        RoleInterface $role = null,
+        ResourceInterface $resource = null,
+        $privilege = null
+    ) {
+        if ($resource instanceof User) {
+            if (null == $this->authUserId) {
+                $authService = new AuthenticationService();
```

Comment:
Wäre es nicht möglich diese Instanziierung stattdessen aus dem Service Manager zu holen? Dann wäre es einfacher, in Tests eine gemockte Version des AuthenticationService reinzuschleusen und die Assertion in einem echten Unit Test zu testen.

Reply:
einverstanden